src/cmake/install.cmake

```diff
@@ -1,56 +1,13 @@
 ###########################################################################
-# Paths for install tree customization.  Note that relative paths are relative
-# to CMAKE_INSTALL_PREFIX.
-set (DEFAULT_BIN_INSTALL_DIR   "bin")
-set (DEFAULT_LIB_INSTALL_DIR   "lib")
-set (DEFAULT_INCLUDE_INSTALL_DIR "include/${PROJECT_NAME}")
-if (UNIX AND NOT SELF_CONTAINED_INSTALL_TREE)
-    # Try to be well-behaved and install into reasonable places according to
-    # the "standard" unix directory heirarchy
-    # TODO: Figure out how to get the correct python directory
-    set (DEFAULT_PYLIB_INSTALL_DIR "lib/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/site-packages")
-    set (DEFAULT_PYLIB3_INSTALL_DIR "lib/python3/site-packages")
-    set (DEFAULT_DOC_INSTALL_DIR "share/doc/${PROJECT_NAME}")
-    set (DEFAULT_MAN_INSTALL_DIR "share/man/man1")
-    set (DEFAULT_FONTS_INSTALL_DIR "share/fonts/${PROJECT_NAME}")
-else ()
-    # Here is the "self-contained install tree" case: the expectation here
-    # is that everything related to this project will go into its own
-    # directory, not into some standard system heirarchy.
-    set (DEFAULT_PYLIB_INSTALL_DIR "python")
-    set (DEFAULT_PYLIB3_INSTALL_DIR "python3")
-    set (DEFAULT_DOC_INSTALL_DIR "doc")
-    set (DEFAULT_MAN_INSTALL_DIR "doc/man")
-    set (DEFAULT_FONTS_INSTALL_DIR "fonts/${PROJECT_NAME}")
-endif ()
-if (EXEC_INSTALL_PREFIX)
-    # Tack on an extra prefix to support multi-arch builds.
-    set (DEFAULT_BIN_INSTALL_DIR   "${EXEC_INSTALL_PREFIX}/${DEFAULT_BIN_INSTALL_DIR}")
-    set (DEFAULT_LIB_INSTALL_DIR   "${EXEC_INSTALL_PREFIX}/${DEFAULT_LIB_INSTALL_DIR}")
-    set (DEFAULT_PYLIB_INSTALL_DIR "${EXEC_INSTALL_PREFIX}/${DEFAULT_PYLIB_INSTALL_DIR}")
-    set (DEFAULT_PYLIB3_INSTALL_DIR "${EXEC_INSTALL_PREFIX}/${DEFAULT_PYLIB3_INSTALL_DIR}")
-    set (DEFAULT_FONTS_INSTALL_DIR "${EXEC_INSTALL_PREFIX}/${DEFAULT_FONTS_INSTALL_DIR}")
-endif ()
-# Set up cmake cache variables corresponding to the defaults deduced above, so
-# that the user can override them as desired:
-set (BIN_INSTALL_DIR ${DEFAULT_BIN_INSTALL_DIR} CACHE STRING
-     "Install location for binaries (relative to CMAKE_INSTALL_PREFIX or absolute)")
-set (LIB_INSTALL_DIR ${DEFAULT_LIB_INSTALL_DIR} CACHE STRING
-     "Install location for libraries (relative to CMAKE_INSTALL_PREFIX or absolute)")
+# Set install paths for the python modules
+# TODO: Figure out how to get the correct python directory
+set (DEFAULT_PYLIB_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/python${PYTHONLIBS_VERSION_MAJOR}.${PYTHONLIBS_VERSION_MINOR}/site-packages")
```

Comment:
I think this line is problematic because PYTHONLIBS_VERSION_MAJOR (and MINOR) are not defined until it reads src/python/CMakeLists.txt, which is after this. So this ends up installing the python modules in a directory named `$prefix/lib/python.` (yes, with the `.` in the path), whereas it used to go in `$prefix/python`. I know that's "wrong" and we want to fix it, but let's confine this review to only using the GNUInstallDirs and moving the locations of fonts and docs (which can't really "break" anything important), and not change anything about the Python location at all, which people need to be prepared for as a breaking change.

Reply:
Maybe, just for this checkin, we change this to

    set (DEFAULT_PYLIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/python")

so that will keep things exactly as they were for this checkin, and then in the other review we can fix all the python stuff once and for all?

Reply:
This was the original setting in the original code, just changed lib to CMAKE_INSTALL_LIBDIR:
https://github.com/OpenImageIO/oiio/blob/master/src/cmake/install.cmake#L11
Did this work before? I didn't even notice this when I made the change, so this is a really good catch. Was this a bug then?

Would it be good to use set (DEFAULT_PYLIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/python${PYTHON_VERSION}/site-packages") since it's set at compile time? Or not?

In the new PR, I have a way around this. :)

Reply:
I looked at the 1.7 source and it's this:
set (DEFAULT_PYLIB_INSTALL_DIR "lib/python/site-packages")
So, I'll change it to that since that's the old default.